variables.tf

```diff
@@ -1,24 +1,383 @@
+    disk_size_gb         = 128
+  }
+}
+
+variable "az_cluster_vm_source_image_reference" {
+  description = "Azure host source image reference"
+  type = object({
+    publisher = string
+    offer     = string
+    sku       = string
+    version   = string
+  })
+
+  default = {
+    publisher = "RedHat"
+    offer     = "RHEL"
+    sku       = "7-LVM"
+    version   = "latest"
+  }
+}
+
+variable "location_instance_type" {
+  description = "The type of azure instance to create"
+  type        = string
+  default     = "Standard_D4s_v3"
+}
+
+variable "cluster_instance_type" {
+  description = "The type of azure instance to create"
+  type        = string
+  default     = "Standard_D4s_v3"
+}
+
+# ##################################################
+# # IBMCLOUD Satellite Location Variables
+# ##################################################
+
+variable "location" {
+  description = "Location Name"
+  default     = "satellite-azure"
+
+  validation {
+    condition     = var.location != "" && length(var.location) <= 32
+    error_message = "Sorry, please provide value for location_name variable or check the length of name it should be less than 32 chars."
+  }
+}
+variable "is_location_exist" {
+  description = "Determines if the location has to be created or not"
   type        = bool
   default     = false
 }
 
-variable "at_create_instance" {
-  description = "Activity Tracker: Controls if Activity Tracker should be created"
+variable "managed_from" {
+  description = "The IBM Cloud region to manage your Satellite location from. Choose a region close to your on-prem data center for better performance."
+  type        = string
+  default     = "wdc"
+}
+
+variable "location_zones" {
+  description = "Allocate your hosts across these three zones"
+  type        = list(string)
+  default     = ["eastus-1", "eastus-2", "eastus-3"]
+}
+
+variable "location_bucket" {
+  description = "COS bucket name"
+  default     = ""
+}
+
+variable "host_labels" {
+  description = "Labels to add to attach host script"
+  type        = list(string)
+  default     = ["env:prod"]
+
+  validation {
+    condition     = can([for s in var.host_labels : regex("^[a-zA-Z0-9:]+$", s)])
+    error_message = "Label must be of the form `key:value`."
+  }
+}
+
+##################################################
+# IBMCLOUD ROKS Cluster Variables
+##################################################
+
+variable "create_cluster" {
+  description = "Create Cluster"
+  type        = bool
+  default     = false
+}
+
+variable "cluster" {
+  description = "Cluster name"
+  type        = string
+  default     = "satellite-ibm-cluster"
+
+  validation {
+    error_message = "Cluster name must begin and end with a letter and contain only letters, numbers, and - characters."
+    condition     = can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.cluster))
+  }
+}
+
+variable "kube_version" {
+  description = "Satellite Kube Version"
+  default     = "4.7_openshift"
```

Comment:
Can we remove default for kube version because that particualr version may be deprecate after some days

Reply:
Removed default value.